components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -583,56 +602,131 @@ void AdsServiceImpl::Shutdown() {
-    VLOG(1) << "Deleting " << rewards_service_base_path.value();
+  return true;
+}
+
+bool RenameAndEncryptState(
+    const base::FilePath& base_path) {
+  const std::vector<std::string> filenames = {
+    "ad_conversions.json",
+    "catalog.json",
+    "client.json",
+    "confirmations.json",
+    "notifications.json"
+  };
+
+  const base::FilePath ads_service_path = base_path.AppendASCII("ads_service");
 
-    if (!base::DeleteFile(rewards_service_base_path)) {
-      VLOG(0) << "Failed to delete " << rewards_service_base_path.value();
+  // Rename files
+  for (const auto& filename : filenames) {
+    const base::FilePath path = ads_service_path.AppendASCII(filename);
+    if (!base::PathExists(path)) {
+      continue;
+    }
+
+    if (!base::Move(path, path.RemoveExtension())) {
+      return false;
+    }
+  }
+
+  // Encrypt files
+  for (const auto& filename : filenames) {
+    const base::FilePath path =
+        ads_service_path.AppendASCII(filename).RemoveExtension();
+
+    std::string value;
+    if (!base::ReadFileToString(path, &value)) {
+      continue;
+    }
+
+    std::string decrypted_value;
+    if (OSCrypt::DecryptString(value, &decrypted_value)) {
+      // File was already migrated
+      continue;
+    }
+
+    std::string encrypted_value;
+    if (!OSCrypt::EncryptString(value, &encrypted_value)) {
+      return false;
+    }
+
+    std::string base64_encrypted_value;
+    base::Base64Encode(encrypted_value, &base64_encrypted_value);
+
+    if (!base::WriteFile(path, base64_encrypted_value)) {
+      return false;
     }
   }
 
   return true;
 }
 
-void AdsServiceImpl::MaybeInitialize() {
-  const base::FilePath path = profile_->GetPath();
+bool MigrateStateOnFileTaskRunner(
+    const base::FilePath& base_path) {
+  if (!MigrateConfirmationsStateFromRewardsServiceToAdsService(base_path)) {
+    return false;
+  }
+
+  if (!RenameAndEncryptState(base_path)) {
+    return false;
+  }
+
+  return true;
+}
+
+void AdsServiceImpl::MigrateState() {
+  VLOG(1) << "Migrating state";
 
   base::PostTaskAndReplyWithResult(
       file_task_runner_.get(), FROM_HERE,
-      base::BindOnce(&MigrateConfirmationsStateOnFileTaskRunner, path),
-      base::BindOnce(&AdsServiceImpl::OnMigrateConfirmationsState,
+      base::BindOnce(&MigrateStateOnFileTaskRunner, profile_->GetPath()),
+      base::BindOnce(&AdsServiceImpl::OnMigrateState,
           AsWeakPtr()));
 }
 
-void AdsServiceImpl::OnMigrateConfirmationsState(
+void AdsServiceImpl::OnMigrateState(
     const bool success) {
   if (!success) {
-    VLOG(0) << "Failed to migrate confirmations state";
+    VLOG(0) << "Failed to migrate state";
     return;
   }
 
-  VLOG(1) << "Successfully migrated confirmations state";
+  SetBooleanPref(prefs::kShouldMigrateState, false);
+
+  VLOG(1) << "Successfully migrated state";
+
+  Initialize();
+}
+
+void AdsServiceImpl::MaybeInitialize() {
+  if (GetBooleanPref(prefs::kShouldMigrateState)) {
```

Comment:
We could perhaps check `Profile::IsNewProfile` and setting `kShouldMigrateState` to `true` to avoid the first run task runner extra IO operations.
Noting that `Profile` shouldn't be used within `components` at all though. I know this PR doesn't introduce that, but we could pass in whether it's a new profile or not.